packages/jest-mock-web-client/src/index.ts

```diff
@@ -0,0 +1,338 @@
+ * copied from it's source, and modified to expose a mockApi instead of it's
+ * actual call. When modifying this file, please keep it in line with it's source.
+ *
+ * Source File: https://github.com/slackapi/node-slack-sdk/blob/8266961af5657ce82d63a410a946a49d29e2bafa/packages/web-api/src/WebClient.ts#L273
+ *
+ * Disabling prefer-default-export as jest doesn't like modules, but typescript does
+ */
+export class MockWebClient {
+  /**
+   * api method family
+   */
+  public readonly api = {
+    test: mockApi(),
+  };
+
+  /**
+   * auth method family
+   */
+  public readonly auth = {
+    revoke: mockApi(),
+    test: mockApi(),
+  };
+
+  /**
+   * bots method family
+   */
+  public readonly bots = {
+    info: mockApi(),
+  };
+
+  /**
+   * channels method family
+   */
+  public readonly channels = {
+    archive: mockApi(),
+    create: mockApi(),
+    history: mockApi(),
+    info: mockApi(),
+    invite: mockApi(),
+    join: mockApi(),
+    kick: mockApi(),
+    leave: mockApi(),
+    list: mockApi(),
+    mark: mockApi(),
+    rename: mockApi(),
+    replies: mockApi(),
+    setPurpose: mockApi(),
+    setTopic: mockApi(),
+    unarchive: mockApi(),
+  };
+
+  /**
+   * chat method family
+   */
+  public readonly chat = {
+    delete: mockApi(),
+    deleteScheduledMessage: mockApi(),
+    getPermalink: mockApi(),
+    meMessage: mockApi(),
+    postEphemeral: mockApi(),
+    postMessage: mockApi(),
+    scheduleMessage: mockApi(),
+    scheduledMessages: {
+      list: mockApi(),
+    },
+    unfurl: mockApi(),
+    update: mockApi(),
+  };
+
+  /**
+   * conversations method family
+   */
+  public readonly conversations = {
+    archive: mockApi(),
+    close: mockApi(),
+    create: mockApi(),
+    history: mockApi(),
+    info: mockApi(),
+    invite: mockApi(),
+    join: mockApi(),
+    kick: mockApi(),
+    leave: mockApi(),
+    list: mockApi(),
+    members: mockApi(),
+    open: mockApi(),
+    rename: mockApi(),
+    replies: mockApi(),
+    setPurpose: mockApi(),
+    setTopic: mockApi(),
+    unarchive: mockApi(),
+  };
+
+  /**
+   * dialog method family
+   */
+  public readonly dialog = {
+    open: mockApi(),
+  };
+
+  /**
+   * dnd method family
+   */
+  public readonly dnd = {
+    endDnd: mockApi(),
+    endSnooze: mockApi(),
+    info: mockApi(),
+    setSnooze: mockApi(),
+    teamInfo: mockApi(),
+  };
+
+  /**
+   * emoji method family
+   */
+  public readonly emoji = {
+    list: mockApi(),
+  };
+
+  /**
+   * files method family
+   */
+  public readonly files = {
+    delete: mockApi(),
+    info: mockApi(),
+    list: mockApi(),
+    revokePublicURL: mockApi(),
+    sharedPublicURL: mockApi(),
+    upload: mockApi(),
+    comments: {
+      delete: mockApi(),
+    },
+    remote: {
+      info: mockApi(),
+      list: mockApi(),
+      add: mockApi(),
+      update: mockApi(),
+      remove: mockApi(),
+      share: mockApi(),
+    },
```

Comment:
worth making a PR over here for the remote block? seems to be missing.

https://github.com/slackapi/node-slack-sdk/blob/8266961af5657ce82d63a410a946a49d29e2bafa/packages/web-api/src/WebClient.ts#L388

Reply:
You're looking at the wrong tree. I'm update the link to be `master`. I also found a few new ones like `admin` that I'm adding

Reply:
Yeah, I thought I used the link in this file? cool though. skimmed it, looks like you got them all.